src/structures/sgx.rs

```diff
@@ -0,0 +1,53 @@
+//! Provides architectural structures for Intel SGX.
+
+bitflags::bitflags! {
+    /// Page protection bits
+    #[repr(transparent)]
+    #[non_exhaustive]
+    pub struct PageFlags: u8 {
+        /// Read access
+        const READ = 1 << 0;
+        /// Write access
+        const WRITE = 1 << 1;
+        /// Execution access
+        const EXEC = 1 << 2;
+    }
+}
+
+/// Page type
+#[derive(Copy, Clone, Debug)]
+#[repr(u8)]
+#[non_exhaustive]
+pub enum PageClass {
+    /// SGX Enclave Control Structure (SECS)
+    Secs = 0,
+    /// Thread Control Structure (TCS)
+    Tcs = 1,
+    /// Regular page
+    Reg = 2,
+    /// Version Array (VA) page
+    Va = 3,
```

Comment:
OK, the choice was based on SDM. Although I would keep abbreviated names for SECS and TCS because they are so natural idioms to anyone who has even peeked to SGX. Non-abbreviated version would more confusing for these two. Is this fine for you? I.e. I'd the following renames:

- Reg -> Regular
- Va -> VersionArray

Reply:
Sure!